test-network-function/lifecycle/suite.go

```diff
@@ -121,45 +119,27 @@ func waitForAllDeploymentsReady(namespace string, timeout, pollingPeriod time.Du
 }
 
 // restoreDeployments is the last attempt to restore the original test deployments' replicaCount
-func restoreDeployments(env *config.TestEnvironment, nsDeployments *map[string]dp.DeploymentMap) {
-	for namespace, originalDeployments := range *nsDeployments {
-		// For each deployment in the namespace, get the current replicas and compare.
-		deployments, notReadyDeployments := getDeployments(namespace)
+func restoreDeployments(env *config.TestEnvironment) {
+	for _, deployment := range env.DeploymentsUnderTest {
+		// For each test deployment in the namespace, refresh the current replicas and compare.
+		deployments, notReadyDeployments := getDeployments(deployment.Namespace)
 
 		if len(notReadyDeployments) > 0 {
 			// Wait until the deployment is ready
-			waitForAllDeploymentsReady(namespace, scalingTimeout, scalingPollingPeriod)
+			waitForAllDeploymentsReady(deployment.Namespace, scalingTimeout, scalingPollingPeriod)
 		}
 
-		for originalDeploymentName, originalDeployment := range originalDeployments {
-			deployment := deployments[originalDeploymentName]
-
-			if deployment.Replicas == originalDeployment.Replicas {
-				continue
-			}
+		if deployments[deployment.Name].Replicas != deployment.Replicas {
+			log.Warn("Deployment ", deployment.Name, " replicaCount (", deployment.Replicas, ") needs to be restored.")
 
 			// Try to scale to the original deployment's replicaCount.
-			runScalingTest(namespace, originalDeploymentName, originalDeployment.Replicas)
+			runScalingTest(deployment.Namespace, deployment.Name, deployment.Replicas)
```

Comment:
here also, it's better to get the parameters inside the function.

Reply:
Done.